contracts/debt/IncurDebtV1.sol

```diff
@@ -0,0 +1,458 @@
+
+        borrower.debt = 0;
+        borrower.collateralInSOHM = uint128(depositedCollateralAfterRepay);
+
+        borrower.collateralInGOHM = uint128(IgOHM(gOHM).balanceTo(depositedCollateralAfterRepay));
+        emit DebtPaidWithCollateral(
+            msg.sender,
+            paidDebt,
+            depositedCollateralAfterRepay,
+            borrower.debt,
+            totalOutstandingGlobalDebt
+        );
+    }
+
+    /**
+     * @notice repay debt with collateral and withdraw the accrued earnings to sOHM/gOHM
+     * - msg.sender must be a borrower
+     * - borrower must have outstanding debt
+     * @param _tokenToReceiveExcess amount of OHM to borrow
+     */
+    function repayDebtWithCollateralAndWithdrawTheRest(address _tokenToReceiveExcess) external override {
+        (uint256 depositedCollateralAfterRepay, uint256 paidDebt) = _repay(msg.sender);
+        assignBorrowerInfoToZero(msg.sender);
+
+        if (depositedCollateralAfterRepay > 0) {
+            if (_tokenToReceiveExcess == sOHM) {
+                IERC20(sOHM).safeTransfer(msg.sender, depositedCollateralAfterRepay);
+            } else if (_tokenToReceiveExcess == gOHM) {
+                IERC20(sOHM).approve(staking, depositedCollateralAfterRepay);
+                IStaking(staking).wrap(msg.sender, depositedCollateralAfterRepay);
+            }
+        }
+
+        emit DebtPaidWithCollateralAndWithdrawTheRest(
+            msg.sender,
+            paidDebt,
+            borrowers[msg.sender].collateralInSOHM,
+            borrowers[msg.sender].debt,
+            totalOutstandingGlobalDebt,
+            depositedCollateralAfterRepay
+        );
+    }
+
+    /**
+     * @notice deposits OHM to pay debt
+     * - msg.sender must be a borrower
+     * - msg.sender's OHM allowance must be >= _ohmAmount
+     * - borrower must have outstanding debt
+     * @param _ohmAmount amount of OHM to borrow
+     */
+    function repayDebtWithOHM(uint256 _ohmAmount) external override isBorrower(msg.sender) {
+        Borrower storage borrower = borrowers[msg.sender];
+
+        if (borrower.debt == 0) revert IncurDebtV1_BorrowerHasNoOutstandingDebt(msg.sender);
+        IERC20(OHM).safeTransferFrom(msg.sender, address(this), _ohmAmount);
+
+        totalOutstandingGlobalDebt -= _ohmAmount;
+        borrower.debt = uint128(borrower.debt - _ohmAmount);
+
+        IERC20(OHM).approve(treasury, _ohmAmount);
+        ITreasury(treasury).repayDebtWithOHM(_ohmAmount);
+
+        emit DebtPaidWithOHM(msg.sender, _ohmAmount, borrower.debt, totalOutstandingGlobalDebt);
+    }
+
+    /**
+     * @notice repays debt using collateral and returns remaining tokens to borrower
+     * - onlyOwner (or governance)
+     * - sends remaining tokens to owner in sOHM
+     * @param _borrower the address that will interact with contract
+     * @param _to where to send remaining sOHM
+     */
+    function forceRepay(address _borrower, address _to) external override onlyGovernor {
```

Comment:
This should probably not be able to have an address passed in for `_to`. Should just return the collateral to the borrower.

Reply:
Fixed.